### PR TITLE
[CONTINT-3593] Add dummy workload to the aks scenario

### PR DESCRIPTION
**Pull request by AliDatadog**
What does this PR do?
---------------------
This PR adds a dummy workload to the aks scenario.

Which scenarios this will impact?
-------------------
az/aks

Motivation
----------
This is a first step towards implementing kata e2e tests.

Additional Notes
----------------

### CODE REVIEW

File: components/datadog/agent/helm.go [hunk 1]
```diff
@@ -197,6 +197,9 @@ func buildLinuxHelmValues(installName, agentImagePath, agentImageTag, clusterAge
 			"prometheusScrape": pulumi.Map{
 				"enabled": pulumi.Bool(true),
 			},
+			"kubelet": pulumi.Map{
+				"tlsVerify": pulumi.Bool(false),
+			},
```
Comment (clamoriniere) [inline components/datadog/agent/helm.go:202]: can we have this setting optionnel? 
IMO we should disable `tlsVerify` only if the k8s destribution doesn't have have a proper certificat, but by default consider that the certificat is valid.
Also it seems that for AKS this setting is not anymore needed see: https://docs.datadoghq.com/containers/kubernetes/distributions?tab=helm#aks-kubelet-certificate

Comment (L3n41c) [reply components/datadog/agent/helm.go]: I agree that `kubelet.tlsVerify: false` shouldn’t be the default.
It should be needed only for Kind.

File: common/utils/kubernetes.go [hunk 2]
```diff
@@ -2,17 +2,24 @@ package utils
...
 	"encoding/json"
-
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"gopkg.in/yaml.v3"
 )
 
 func KubeconfigToJSON(kubeConfig pulumi.Output) pulumi.StringInput {
-	return kubeConfig.ApplyT(func(config interface{}) (string, error) {
-		jsonConfig, err := json.Marshal(config)
+	return kubeConfig.ApplyT(func(config string) (string, error) {
```
Comment (L3n41c) [inline common/utils/kubernetes.go:10]: This changes the signature of the `KubeconfigToJSON` function.
Won’t it break all the [callers](https://github.com/DataDog/test-infra-definitions/blob/c45b6947c391d5d70eac6fde60e1711b88925e0c/scenarios/aws/eks/run.go#L169) that are not updated in this PR ?

File: components/datadog/agent/helm.go (hunk 1 above)
Comment (AliDatadog) [reply components/datadog/agent/helm.go]: Thank you for the info. I found a way to add environment-specific fields in the helm chart. It's done in this commit: 60cdcefbccd5e45c555fcf82c508456661f8a041.

File: scenarios/azure/aks/run.go [hunk 3]
```diff
@@ -13,11 +23,82 @@ func Run(ctx *pulumi.Context) error {
...
 
-	_, kubeConfig, err := aks.NewCluster(env, "aks", nil)
+	cluster, kubeConfig, err := aks.NewCluster(env, "aks", nil)
 	if err != nil {
 		return err
 	}
 
+	// Export cluster’s properties
 	ctx.Export("kubeconfig", kubeConfig)
+	ctx.Export("kube-cluster-name", cluster.Name.ApplyT(func(value string) (string, error) { return value, nil }))
```
Comment (L3n41c) [inline scenarios/azure/aks/run.go:33]: Why do we need an `ApplyT` here ?
The function looks like identity.

```suggestion
	ctx.Export("kube-cluster-name", cluster.Name)
```

File: common/utils/kubernetes.go (hunk 2 above)
Comment (AliDatadog) [reply common/utils/kubernetes.go:10]: I think the function name wasn't changed. The only change should be that we unmarshal as yaml first.

Comment (vboulineau) [reply common/utils/kubernetes.go:11]: Your change assumes that `kubeConfig` is a YAML string.
In this case, you should change the signature to take in a ` kubeConfigYAML pulumi.StringOutput` and perhaps change the name of the function to `KubeconfigYAMLToJSON` as it does not do exactly what it says.

File: scenarios/azure/aks/run.go [hunk 4]
```diff
@@ -13,11 +23,98 @@ func Run(ctx *pulumi.Context) error {
...
 	}
 
-	_, kubeConfig, err := aks.NewCluster(env, "aks", nil)
+	cluster, kubeConfig, err := aks.NewCluster(env, "aks", nil)
 	if err != nil {
 		return err
 	}
 
+	// Export cluster’s properties
 	ctx.Export("kubeconfig", kubeConfig)
```
Comment (vboulineau) [inline scenarios/azure/aks/run.go:32]: I'd suggest to pu these modifications after https://github.com/DataDog/test-infra-definitions/pull/484 to directly implement the Kubernetes component for AKS.

Comment (AliDatadog) [reply scenarios/azure/aks/run.go:32]: Do you have an estimation of when it should be merged ?

File: common/utils/kubernetes.go (hunk 2 above)
Comment (AliDatadog) [reply common/utils/kubernetes.go:11]: I did it in the last commit

File: scenarios/azure/aks/run.go [hunk 5]
```diff
@@ -13,11 +25,105 @@ func Run(ctx *pulumi.Context) error {
...
 	}
 
-	_, kubeConfig, err := aks.NewCluster(env, "aks", nil)
+	clusterComp, err := components.NewComponent(*env.CommonEnvironment, env.Namer.ResourceName("aks"), func(comp *kubeComp.Cluster) error {
+		cluster, kubeConfig, err := aks.NewCluster(env, "aks", nil)
+		if err != nil {
+			return err
+		}
+
+		ctx.Export("kubeconfig", kubeConfig)
```
Comment (vboulineau) [inline scenarios/azure/aks/run.go:34]: Not necessary anymore, you will export `clusterComp` later

File: scenarios/azure/aks/run.go [hunk 6]
```diff
@@ -13,11 +25,105 @@ func Run(ctx *pulumi.Context) error {
...
+		// Filling Kubernetes component from EKS cluster
+		comp.ClusterName = cluster.Name
+		comp.KubeConfig = kubeConfig
+
+		// Building Kubernetes provider
+		aksKubeProvider, err := kubernetes.NewProvider(env.Ctx, env.Namer.ResourceName("k8s-provider"), &kubernetes.ProviderArgs{
```
Comment (vboulineau) [inline scenarios/azure/aks/run.go:40]: The provider needs to be store inside `comp`.

Comment (AliDatadog) [reply scenarios/azure/aks/run.go:39]: I didn't see it for other scenarios but I added it now

File: go.mod [hunk 7]
```diff
@@ -35,7 +35,7 @@ require (
 require (
 	dario.cat/mergo v1.0.0 // indirect
 	github.com/Microsoft/go-winio v0.6.1 // indirect
-	github.com/ProtonMail/go-crypto v1.0.0 // indirect
+	github.com/ProtonMail/go-crypto v0.0.0-20230828082145-3c4c8a2d2371 // indirect
```
Comment (vboulineau) [inline go.mod:38]: I guess it should not be present

Comment (AliDatadog) [reply go.mod]: I bumped it again to v1.0.0 👍

**Review by vboulineau — APPROVED** [review]

**Pull request merged**